src/components/input/Select.tsx

```diff
@@ -112,22 +126,37 @@ function SelectOption<T>({
         },
         classes,
       )}
-      onClick={() => {
-        if (!disabled) {
-          selectOrToggle();
+      onClick={e => {
+        // Do not invoke callback if clicked element is the checkbox, as it has
+        // its own event handler.
+        if (!disabled && e.target !== checkboxRef.current) {
+          selectOneValue();
         }
       }}
-      onKeyPress={e => {
-        if (!disabled && ['Enter', 'Space'].includes(e.code)) {
+      onKeyDown={e => {
+        if (disabled) {
+          return;
+        }
+
+        if (
+          ['Enter', 'Space'].includes(e.key) &&
```

Comment:
```suggestion
          ['Enter', ' '].includes(e.key) &&
```

For `key` the value is the printable representation, if one exists.

Reply:
Oops. I quickly tested only with <kbd>Enter</kbd>, and missed this.